tensorflow/compiler/xla/service/gpu/nvptx_compiler.h

```diff
@@ -35,6 +36,41 @@ void WarnIfBadDriverJITVersion();
 // Returns the directory containing nvvm libdevice files.
 string GetLibdeviceDir(const HloModuleConfig& hlo_module_config);
 
+/*
+Persistent compilation cache.
+This cache store .ptx and .cubin files to be used by subsequent compilations.
+The cache is a directory of files. The file name is a hash of the file content.
+All file are read (disk IO) and stored in memory when the cache is consructed.
+When an uncached compilation occurs, the result is written (disk IO) to the
+cache directory immediately. Autotuning is currently non-deterministic, so a
+few executions might be required to populate the cache. 
+
+Deployemt:
+For best performance, keep the cache small (per model) containing only the
+binaries needed for this execution. In that scenario, after cache creation,
+there will be no disk IO.
+*/
+class persistentCompilationCache
+{
+    static const int64 ptx_hash_ = 0xBA55ED50;
```

Comment:
`kPtxHash` is more idiomatic.  Please also add a comment on what this means or use a more descriptive name.  And can this be `constexpr`?

Reply:
done